pymeasure/adapters/serial.py

```diff
@@ -103,9 +103,23 @@ def _read_bytes(self, count, break_on_termchar, **kwargs):
             return self.connection.read_until(self.read_termination.encode(),
                                               count if count > 0 else None,
                                               **kwargs)
+        elif count >= 0:
+            return self.connection.read(count, **kwargs)
         else:
-            # At -1 we read a very large number of bytes, which can be considered the whole buffer.
-            return self.connection.read(1e99 if count == -1 else count, **kwargs)
+            # For -1 we empty the buffer completely
+            return self._read_bytes_until_timeout()
+
+    def _read_bytes_until_timeout(self, **kwargs):
+        """Read from the serial until a timeout occurs, regardless of how many bytes."""
+        # `Serial.readlines()` sounds appropriate instead but it turns out it has an
+        # unpredictable timeout
+        data = bytes()
+        chunk_size = 256
```

Comment:
IMO, this should be a kwarg  of the method -- folks might want to tweak this for performance optimisation/tradeoffs.

Reply:
Sure, no reason not to expose it.